tyk-data-plane/values.yaml

```diff
@@ -0,0 +1,382 @@
+    affinity: {}
+
+    # extraEnvs is used to set gateway env variables
+    # - name: TYK_GW_HTTPSERVEROPTIONS_SSLINSECURESKIPVERIFY
+    #   value: "true"
+    extraEnvsFromSecret:
+      - name: TYK_GW_SLAVEOPTIONS_RPCKEY
+        valueFrom:
+          secretKeyRef:
+            name: secrets-tyk-data-plane
+            key: rpcKey
+      - name: TYK_GW_SLAVEOPTIONS_APIKEY
+        valueFrom:
+          secretKeyRef:
+            name: secrets-tyk-data-plane
+            key: apiKey
+      - name: TYK_GW_SECRET
+        valueFrom:
+          secretKeyRef:
+            name: secrets-tyk-data-plane
+            key: APISecret
+      - name: TYK_GW_SLAVEOPTIONS_GROUPID
+        valueFrom:
+          secretKeyRef:
+            name: secrets-tyk-data-plane
+            key: groupID
+    extraEnvsFromConfigMap:
+      - name: TYK_GW_SLAVEOPTIONS_CONNECTIONSTRING
+        valueFrom:
+          configMapKeyRef:
+            name: config-tyk-data-plane
+            key: rpcConnString
+      - name: TYK_GW_SLAVEOPTIONS_USESSL
+        valueFrom:
+          configMapKeyRef:
+            name: config-tyk-data-plane
+            key: rpcUseSSL
+      - name: TYK_GW_SLAVEOPTIONS_SSLINSECURESKIPVERIFY
+        valueFrom:
+          configMapKeyRef:
+            name: config-tyk-data-plane
+            key: rpcSslInsecureSkipVerify
+
+
+
+
+
+    # used to mount files to the gateway containers
+    #  - name: test
+    #    filename: test.txt
+    #    mountPath: /opt/test
+    mounts: []
+
+
+
+tyk-pump:
+  ## Default values for tyk-pump chart.
+  ## This is a YAML-formatted file.
+  ## Declare variables to be passed into your templates.
+  ## See Tyk Helm documentation for installation details:
+  ## https://tyk.io/docs/tyk-oss/ce-helm-chart/
+  ## Registry for all Tyk images - https://hub.docker.com/u/tykio
+
+  # Overrides chart name. It is truncated to 63 characters.
+  # Default value: tyk-pump.name
+  nameOverride: ""
+
+  # Overrides app name. IT is truncated to 63 characters.
+  # Default value: tyk-pump.fullname
+  fullnameOverride: ""
+
+  # If pump is enabled the Gateway will create and collect analytics data to send
+  # to a data store of your choice. These can be set up in the pump config. The
+  # possible pump configs can be found here:
+  # https://github.com/TykTechnologies/tyk-pump#configuration
+  pump:
+
+    # number for replicas for pump deployment
+    replicaCount: 1
+
+    image:
+      # image repository for Tyk pump
+      repository: docker.tyk.io/tyk-pump/tyk-pump
+
+      # tag for Tyk pump
+      tag: v1.7.0
+
+      # image pull policy
+      pullPolicy: IfNotPresent
+
+    # Choose the pump backend for Tyk Pump. ["", "mongo", "postgres", "prometheus"]
+    # It defines the pumps to be created by default.
+    # If you would like to use other backends such as ElasticSearch, please set this field to "" and configure
+    # your Tyk Pump via environment variables.
+    backend: "prometheus"
```

Comment:
Just reminder, this part is highly likely to change based on #21. if #21 is merged before, we should update this part too.

Reply:
Ok, I will wait for it to be merged and modify it accordingly.